### PR TITLE
[INTEL MKL] Fixing inter op default setting in intel.

### PR DESCRIPTION
**Pull request by Srini511**
Removing openmp calls in mkl default inter op settings due to side effects.

### CODE REVIEW

File: tensorflow/core/common_runtime/process_util.cc [hunk 1]
```diff
@@ -97,18 +97,39 @@ int32 NumIntraOpThreadsFromEnvironment() {
   return (val && strings::safe_strto32(val, &num)) ? num : 0;
 }
 
+#ifdef INTEL_MKL
+int32 OMPThreadsFromEnvironment() {
+  int32 num;
+  const char* val = std::getenv("OMP_NUM_THREADS");
```
Comment (penpornk) [inline tensorflow/core/common_runtime/process_util.cc]: Please add a comment saying that 
1) `std::getenv` is thread-safe (as long as no other function modifies the host environment) from C++11 onward.
2) Most of TF code (except tests and experimental code) doesn't call `setenv` and `unsetenv`.

File: tensorflow/core/common_runtime/process_util.cc [hunk 2]
```diff
@@ -139,7 +160,7 @@ void SchedClosure(std::function<void()> closure) {
   uint64 id = tracing::GetUniqueArg();
   tracing::RecordEvent(tracing::EventCategory::kScheduleClosure, id);
 
-  Env::Default()->SchedClosure([id, closure = std::move(closure)]() {
+  Env::Default()->SchedClosure([ id, closure = std::move(closure) ]() {
```
Comment (penpornk) [inline tensorflow/core/common_runtime/process_util.cc]: Just curious. Are the extra spaces the formatter's doing?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/common_runtime/process_util.cc (hunk 2 above)
Comment (Srini511) [reply tensorflow/core/common_runtime/process_util.cc]: yes.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request merged**